src/metabase/lib/schema/id.cljc

```diff
@@ -0,0 +1,16 @@
+(ns metabase.lib.schema.id
+  (:require
+   [metabase.lib.schema.common :as common]
+   [metabase.util.malli.registry :as mr]))
+
+;;; these aren't anything special right now, but maybe in the future we can do something special/intelligent with
+;;; them, e.g. when we start working on the generative stuff
+
+(mr/def ::database
```

Comment:
I was in the yaml spec checker recently where all of the specs are autoresolved namespace keywords. It really makes it difficult to find things. It's a sharp edge to using so many small namespaces with autoresolved keywords like this.

Reply:
I actually came to the opposite conclusion, I initially went with using `:mbql/` as the prefix for everything, but then it's a problem because

- It's not obvious what namespace the definition is defined in
- There is nothing to encourage you load the namespace it's defined in. You either have to remember to do it and do something hacky like `(comment important.namespace.with-schemas/keep-me)` or cross your fingers and hope someone else did it for you 
- It's a lot easier to accidentally stomp on things defined elsewhere

So I think there are pros and cons to either approach. But If the consensus is that the cons for namespaced keywords outweigh the benefits I'll change it back

Reply:
I'm not 100% sure about this approach TBH, I think having to go `:require` all the namespaces is a little bit annoying. But at this point I'm leaning towards getting this in and possibly reconsidering in the near future